content/blog/readr-2-0-0/index.Rmd

```diff
@@ -0,0 +1,265 @@
+
+```{r, message = FALSE}
+read_tsv("flights_AA.tsv", name_repair = janitor::make_clean_names)
+
+read_tsv("flights_AA.tsv", name_repair = ~ janitor::make_clean_names(., case = "lower_camel"))
+```
+
+```{r, show = FALSE}
+unlink(files)
+```
+
+## UTF-16 and UTF-32 support
+
+Edition two now has much better support for UTF-16 and UTF-32 multi-byte unicode encodings.
+When files with these encodings are read they are automatically converted to UTF-8 internally in an efficient streaming fashion.
+
+## Lazy reading
+
+Edition two uses lazy reading by default.
+When you first call a `read_*()` function the delimiters and newlines throughout the entire file are found, but the data is not actually read until it is used in your program.
+This can provide substantial speed improvements for reading character data.
+It is particularly useful during interactive exploration of only a subset of a full dataset.
+
+However this also means that problematic values are not necessarily seen
+immediately, only when they are actually read.
+Because of this a warning will be issued the first time a problem is encountered,
+which may happen after initial reading.
+
+Run `problems()` on your dataset to read the entire dataset and return all of the problems found.
+Run `problems(lazy = TRUE)` if you only want to retrieve the problems found so far.
+
+Deleting files after reading is also impacted by laziness.
+On Windows open files cannot be deleted as long as a process has the file open.
+Because readr keeps a file open when reading lazily this means you cannot read, then immediately delete the file.
+readr will in most cases close the file once it has been completely read.
+However, if you know you want to be able to delete the file after reading it is best to pass `lazy = FALSE` when reading the file.
+
+## Control over quoting and escaping when writing
+
+You can now explicitly control how fields are quoted and escaped when writing with the `quote` and `escape` arguments to `write_*()` functions.
+
+`quote` has three options.
+
+1. 'needed' - which will quote fields only when needed.
+2. 'all' - which will always quote all fields.
+3. 'none' - which will never quote any fields.
+
+`escape` also has three options, to control how quote characters are escaped.
+
+1. 'double' - which will use double quotes to escape quotes.
+2. 'backslash' - which will use a backslash to escape quotes.
+3. 'none' - which will not do anything to escape quotes.
+
+We hope these options will give people the flexibility they need when writing files using readr.
+
+## Literal data
+
+In edition one the reading functions treated any input with a newline in it or vectors of length > 1 as literal data.
+In edition two vectors of length > 1 are now assumed to correspond to multiple files.
+Because of this we now have a more explicit way to represent literal data, by putting `I()` around the input.
+
+```{r}
+readr::read_csv(I("a,b\n1,2"))
+```
+
+## Lighter installation requirements
```

Comment:
Maybe lump together some of these smaller changes in a bulleted list under a new heading like "Other minor improvements"